tests/Integration/Behaviour/Features/Context/Domain/CmsPageFeatureContext.php

```diff
@@ -0,0 +1,365 @@
+    {
+        $command = new AddCmsPageCommand(
+            (int) $id,
+            [$this->defaultLangId => 'test'],
+            [$this->defaultLangId => 'test'],
+            [$this->defaultLangId => 'test'],
+            [$this->defaultLangId => 'test'],
+            [$this->defaultLangId => 'test'],
+            [$this->defaultLangId => 'test'],
+            true,
+            true,
+            [$this->defaultShopId]
+        );
+        try {
+            $cmsPageId = $this->getCommandBus()->handle($command);
+            SharedStorage::getStorage($reference, new \CMS((int) $cmsPageId->getValue()));
+
+            throw new NoExceptionAlthoughExpectedException('Cms page creation expected to fail, but it succeeded');
+        } catch (\Exception $e) {
+            $this->latestResult = $e;
+        }
+    }
+
+    /**
+     * @When I edit cms page :reference with following properties:
+     */
+    public function editCmsPage($reference, TableNode $node)
+    {
+        $cmsId = (int) SharedStorage::getStorage()->get($reference)->id;
+        $command = new EditCmsPageCommand($cmsId);
+        $data = $node->getRowsHash();
+
+        if (isset($data['meta_title'])) {
+            $command->setLocalizedTitle([$this->defaultLangId => $data['meta_title']]);
+        }
+        if (isset($data['head_seo_title'])) {
+            $command->setLocalizedMetaTitle([$this->defaultLangId => $data['head_seo_title']]);
+        }
+        if (isset($data['meta_description'])) {
+            $command->setLocalizedMetaDescription([$this->defaultLangId => $data['meta_description']]);
+        }
+        if (isset($data['meta_keywords'])) {
+            $command->setLocalizedMetaKeyword([$this->defaultLangId => $data['meta_keywords']]);
+        }
+        if (isset($data['link_rewrite'])) {
+            $command->setLocalizedFriendlyUrl([$this->defaultLangId => $data['link_rewrite']]);
+        }
+        if (isset($data['content'])) {
+            $command->setLocalizedContent([$this->defaultLangId => $data['content']]);
+        }
+        if (isset($data['indexation'])) {
+            $command->setIsIndexedForSearch((bool) $data['indexation']);
+        }
+        if (isset($data['active'])) {
+            $command->setIsDisplayed((bool) $data['active']);
+        }
+
+        $this->getCommandBus()->handle($command);
+        SharedStorage::getStorage()->set($reference, new \CMS($cmsId));
+    }
+
+    /**
+     * @When I attempt to edit cms page :reference providing illegal content value :value
+     */
+    public function editCmsPageProvidingIllegalContent($reference, $value)
+    {
+        $cmsId = (int) SharedStorage::getStorage()->get($reference)->id;
+        $command = new EditCmsPageCommand($cmsId);
+        $command->setLocalizedContent(
+            [$this->defaultLangId => $value]
+        );
+        try {
+            $this->getCommandBus()->handle($command);
+            SharedStorage::getStorage()->set($reference, new \CMS($cmsId));
```

Comment:
If you expect exception and `CMS` is alkready in shared storage, what is the purpose of storing it here again?